test/24-contracts.ts

```diff
@@ -0,0 +1,212 @@
+import { assert, expect } from 'chai';
+
+import * as semver from 'balena-semver';
+
+import * as constants from '../src/lib/constants';
+import { containerContractsFulfilled } from '../src/lib/contracts';
+import * as osRelease from '../src/lib/os-release';
+import supervisorVersion = require('../src/lib/supervisor-version');
+
+describe('Container contracts', () => {
+	// Because the supervisor version will change whenever the
+	// package.json will, we generate values which are above
+	// and below the current value, and use these to reason
+	// about the contract engine results
+	const supervisorVersionGreater = `${semver.major(supervisorVersion)! +
+		1}.0.0`;
+	const supervisorVersionLesser = `${semver.major(supervisorVersion)! - 1}.0.0`;
+
+	before(async () => {
+		// We ensure that the versions we're using for testing
+		// are the same as the time of implementation, otherwise
+		// these tests could fail or succeed when they shouldn't
+		expect(await osRelease.getOSSemver(constants.hostOSVersionPath)).to.equal(
+			'2.0.6',
```

Comment:
I think I'm confused here by this version number.
Edit: Ah, no I'm not, I've seen the end of this commit list now.